backends/consul_client.go

```diff
@@ -0,0 +1,94 @@
+package backends
+
+import (
+	"crypto/tls"
+	"crypto/x509"
+	"io/ioutil"
+	"net/http"
+	"path"
+	"strings"
+
+	"github.com/hashicorp/consul/api"
+)
+
+// Client provides a wrapper around the consulkv client
+type ConsulClient struct {
+	client *api.KV
+}
+
+// NewConsulClient returns a new client to Consul for the given address
```

Comment:
I took the liberty of moving this to the `github.com/kelseyhightower/confd/backends` package as I was not able to use the great `Config` struct due to a circular import if it was in the `github.com/kelseyhightower/confd/backends/consul` packags. I'm open to making `NewConsulClient` "ugly" again enumerating all the values from the `Config` struct as the function arguments if this move is objectionable.